### DISCUSSION TITLE
Tapo L530 - Does it support transition?

### DISCUSSION THREAD

**Discussion by wutr** [petretiandrea/home-assistant-tapo-p100#673] [Q&A]
As per the title, does the Tapo L530(e) support brightness transition? I've tried calling the light.turn_on service in Home Assistant but changing the transition parameter doesn't make any difference. I've tried it with 'Light On/Off Gradually' enabled and disabled in the device settings in the Tapo app, but again changing the transition parameter in Home Asisstant makes no difference either way. Maybe I'm missing something?

**Comment by petretiandrea** [comment]
Actually, is not supported by integration. If is supported by Tapo App, I can add this new feature

**Reply by wutr** [reply]
Thank you. I can't find it in the Tapo App, so it is probably not supported at the moment.

**Reply by will-roscoe** [reply]
I have a setting in the Tapo app for my L630 rgbw bulbs that toggles turning off lights gradually, found in the device specific settings menu. There doesn't seem to be any granularity to how fast or slow the bulbs transition but I have had luck with changing transition speeds via the Adaptive Lighting custom component on HACS so I assume it is possible somehow. ![image](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/146133682/a78e5285-b299-4242-b612-0cb87e67327e)

**Reply by wutr** [reply]
The adaptive lighting component may do its own transition by sending brightness commands at calculated intervals. This is how I achieve transitions in automations/scripts for lights that don't natively support transitions.

**Comment by jakubigla** [comment]
Is there any update on this? Really disappointed the transition is not supported in the integration :(

**Reply by brickheadbs** [reply]
I'll request this also. Google Home is able to do it, and the tapo app. Google home has wake and sleep features.

**Reply by Bluscream** [reply]
i would love this feature too, especially since i use it as night light and dont want the transition to be noticed
